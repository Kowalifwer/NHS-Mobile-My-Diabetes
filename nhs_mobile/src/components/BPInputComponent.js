import React from 'react';
import {
    View,
    StyleSheet,
    Text,
    TextInput,
    Alert,
    SafeAreaView, 
    ScrollView,
} from 'react-native';
import GlobalStyle from '../styles/GlobalStyle';
import DateTimePicker from '@react-native-community/datetimepicker';
import CustomButton from './CustomButton';


// this is the food input which gets added to the page when user clicks the + button
class BPInputComponent extends React.Component {
    constructor() {
        super();
        this.mounted = false;
        this.state = {
            time: new Date(),
            time_string: "",
            showTimePicker: false,
            dict: {time: "", arm: "", systolic: "", diastolic: ""},
        };
        console.log("ismounted: ", this.isMounted)
    }

    componentDidMount() {
        console.log("COMPONENT MOUNTED");
        this.mounted = true;
    }

    render() {
        return (
            <View>
                {this.state.showTimePicker && (
                    <DateTimePicker
                        testID="timePicker"
                        value={this.state.time}
                        display="default"
                        mode="time"
                        onChange={(event, new_time) => {
                            this.setState({
                                showTimePicker: false,
                                time: new_time,
                                time_string: `${new_time.getHours()}:${new_time.getMinutes()}`,
                            })
                        }}
                    />
                )}
                <CustomButton
                    onPressFunction={() => {
                        console.log("PRESSED BUTTON, IS MOUNTED?: ", this.mounted);
                        this.setState({showTimePicker: true});
                        console.log("showTimePicker: true");
                    }}
                    title="Enter Time"
                    color="#008c8c"
                />
                <TextInput
                    style={GlobalStyle.InputField}
                    placeholder='arm'
                    onChangeText={(value) => {
                        this.dict["arm"] = value.trim()
                    }}
                    multiline={true}
                    numberOfLines={1}
                />
                <TextInput
                    style={GlobalStyle.InputField}
                    placeholder="systolic"
                    keyboardType="numeric"
                    onChangeText={(value) => {
                        this.dict["systolic"] = value.trim()
                    }}
                    multiline={true}
                    numberOfLines={1}
                />
                                <TextInput
                    style={GlobalStyle.InputField}
                    placeholder="diastolic"
                    keyboardType="numeric"
                    onChangeText={(value) => {
                        this.dict["diastolic"] = value.trim()
                    }}
                    multiline={true}
                    numberOfLines={1}
                />
            </View>
        );
    }
}

export default BPInputComponent