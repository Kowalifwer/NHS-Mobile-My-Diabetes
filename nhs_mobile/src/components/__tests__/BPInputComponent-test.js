import React from 'react';
import renderer from 'react-test-renderer';
import BPInputComponent from '../BPInputComponent';

it('renders correctly across screens', () => {
    const tree = renderer.create(<BPInputComponent />).toJSON();
    expect(tree).toMatchSnapshot();
  });