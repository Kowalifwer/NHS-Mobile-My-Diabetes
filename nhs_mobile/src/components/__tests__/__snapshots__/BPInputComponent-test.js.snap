// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly across screens 1`] = `
<View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    hitSlop={
      Object {
        "bottom": 10,
        "left": 10,
        "right": 10,
        "top": 10,
      }
    }
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "#008c8c",
        },
        Object {
          "alignItems": "center",
          "borderRadius": 10,
          "height": "auto",
          "justifyContent": "center",
          "margin": 10,
          "minHeight": 40,
          "padding": 10,
          "width": 350,
        },
        Object {},
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "#ffffff",
          "fontSize": 20,
          "marginTop": -5,
          "textAlign": "center",
        }
      }
    >
      Enter Time
    </Text>
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "blue",
          "fontFamily": "AbrilFatface",
          "fontWeight": "300",
        },
      ]
    }
  >
    Blood Pressure Reading 
    NaN
  </Text>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="arm"
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "#ffffff",
        "borderColor": "#555",
        "borderRadius": 10,
        "borderWidth": 1,
        "fontSize": 20,
        "height": 40,
        "marginBottom": 10,
        "marginTop": 10,
        "paddingBottom": 5,
        "textAlign": "center",
        "width": 350,
      }
    }
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    keyboardType="numeric"
    onChangeText={[Function]}
    placeholder="systolic"
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "#ffffff",
        "borderColor": "#555",
        "borderRadius": 10,
        "borderWidth": 1,
        "fontSize": 20,
        "height": 40,
        "marginBottom": 10,
        "marginTop": 10,
        "paddingBottom": 5,
        "textAlign": "center",
        "width": 350,
      }
    }
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    keyboardType="numeric"
    onChangeText={[Function]}
    placeholder="diastolic"
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "#ffffff",
        "borderColor": "#555",
        "borderRadius": 10,
        "borderWidth": 1,
        "fontSize": 20,
        "height": 40,
        "marginBottom": 10,
        "marginTop": 10,
        "paddingBottom": 5,
        "textAlign": "center",
        "width": 350,
      }
    }
    underlineColorAndroid="transparent"
  />
</View>
`;
